src/options.js

```diff
@@ -0,0 +1,118 @@
+/*global chrome */
+function restore_options() {
+    loadDoNotReplaceList(function(doNotReplaceList) {
+        displayDoNotReplaceList(doNotReplaceList);
+    });
+}
+
+function loadDoNotReplaceList(callback) {
+    chrome.storage.sync.get(
+        {
+            doNotReplaceList: []
+        },
+        function(items) {
+            if (callback) {
+                callback(items.doNotReplaceList);
+            }
+        }
+    );
+}
+
+function saveDoNotReplaceList(doNotReplaceList, callback) {
+    const items = { doNotReplaceList: doNotReplaceList };
+    chrome.storage.sync.set(items, function(items) {
+        if (callback) {
+            callback();
+        }
+    });
+}
+
+function addItem() {
+    const newEntryField = document.getElementById("newEntry");
+
+    // We can't use the URL API here, as it requires a protocol
```

Comment:
Why would there not be a protocol? I think it's because the user might not type one, but I'm not totally sure.